Improve (email, password) inputs of login and register forms

coderNadir opened the pull request:
# PR

- [x] Have you followed the [contributing guidelines](https://github.com/ever-co/ever-gauzy/blob/master/.github/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

---

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ever-co/ever-gauzy?pullRequest=6952) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ever-co/ever-gauzy?pullRequest=6952) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: coderNadir<br/>:x: BM-Nadir<br/><hr/>**BM-Nadir** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ever-co/ever-gauzy?pullRequest=6952) it.</sub>

evereq commented:
@rahul-rocket I like this PR, but can you check if all good? Thanks!

evereq commented:
@adkif / @rahul-rocket can you review this PR too please?

coderNadir commented:
- I've added a RegEx to the password to not let the user enter a space whether at the beginning or at the end and warn him about it.
- I've added a customized Directive to the email input in for (login, register) forms that will always remove the space at the beginning or the end -> why? because if we tried to login with an email that has a space we are going to get an unauthorized response, also the email will be saved with (rememberMe fn) with that extra space and will lead us to the same error

adkif reviewed (approved):
Looks good!
